### PR TITLE
fix secure cookie domain overlay

### PR DESCRIPTION
**Pull request by bagder**
@piru: I assume you don't mind me using your patch here

### CODE REVIEW

**Comment by piru** [comment]
> @piru: I assume you don't mind me using your patch here

I don't mind at all. Fix the typo though: s/exsiting/existing/g

File: tests/data/test1561 [hunk 1]
```diff
@@ -35,12 +35,12 @@ nomnom
...
-Set-Cookie: super=secret; domain=example.com; path=/%TESTNUMBER/; httponly;
-Set-Cookie: super=secret; domain=example.com; path=/15; httponly;
+Set-Cookie: super=plain; domain=example.com; path=/%TESTNUMBER; httponly;
+Set-Cookie: super=plain; domain=example.com; path=/%TESTNUMBER/; httponly;
+Set-Cookie: super=plain; domain=example.com; path=/15; httponly;
 Set-Cookie: public=yes; domain=example.com; path=/foo;
-Set-Cookie: supersuper=secret; domain=example.com; path=/%TESTNUMBER/login/en;
-Set-Cookie: supersuper=secret; domain=example.com; path=/%TESTNUMBER/login;
+Set-Cookie: supersuper=plain; domain=example.com; path=/%TESTNUMBER/login/en;
+Set-Cookie: supersuper=plain; domain=example.com; path=/%TESTNUMBER/login;
```
Comment (danielgustafsson) [inline tests/data/test1561:43]: Why this change?

File: lib/cookie.c [hunk 2]
```diff
@@ -1059,11 +1060,39 @@ Curl_cookie_add(struct Curl_easy *data,
...
+        }
+      }
+    }
+    lastc = clist;
+    clist = clist->next;
+  }
+
+  clist = c->cookies[myhash];
+  replace_old = FALSE;
+  while(clist) {
```
Comment (danielgustafsson) [inline lib/cookie.c:1078]: Traversing clist twice adds a computational cost, might be unavoidable but we should probably at the least add a comment on why this is done.

File: tests/data/test1561 (hunk 1 above)
Comment (bagder) [reply tests/data/test1561:43]: For making the test easier to read. I decided to change the contents *set* in the two responses (HTTPS vs HTTP) just to make it more obvious in the final output which of the same name cookies that actually survive.

File: lib/cookie.c (hunk 2 above)
Comment (bagder) [reply lib/cookie.c:1078]: I'll see if I can work it down to a single loop

Comment (bagder) [reply lib/cookie.c:1078]: Just needed a little massaging. Thanks for highlighting this.
